File: RQ1/CodeForces-Automatic/584A.py
def main():
    # PRE:
    # Input Format:
    # The single line contains two numbers, n and t (1 ≤ n ≤ 100, 2 ≤ t ≤ 10) — the length of the number and the number it should be divisible by.
    # POST:
    # Program Description:
    # Olesya loves numbers consisting of n digits, and Rodion only likes numbers that are divisible by t. Find some number that satisfies both of them.Your task is: given the n and t print an integer strictly larger than zero consisting of n digits that is divisible by t. If such number doesn't exist, print  - 1.
    # Output Format:
    # Print one such positive number without leading zeroes, — the answer to the problem, or  - 1, if such number doesn't exist. If there are multiple possible answers, you are allowed to print any of them.
    n, t = map(int, input().split())
    i = int('1' +'0' * (n - 1))
    j = int('1' +'0' * n)
    
    ans = 0
    for num in range(i, j):
          if num % t == 0:
                ans = num
                break
    
    print(-1) if ans == 0 else print(ans)

File: RQ1/CodeForces-Automatic/584A.c
#include <stdio.h>
#include <assert.h>

int main()
{
    int n, t;
    scanf("%d %d", &n, &t);
    
    assert(n >= 1 && n <= 100); // PRE: n must be between 1 and 100
    assert(t >= 2 && t <= 10); // PRE: t must be between 2 and 10
    
    char s[101] = {0};

    if (t == 10)
    {
        if (n == 1)
        {
            printf("-1\n");
            return 0;
        }
        s[0] = '1';
        for (int i = 1; i < n; i++)
        {
            s[i] = '0';
        }
    }
    else
    {
        char c = t + '0';
        for (int i = 0; i < n; i++)
        {
            s[i] = c;
        }
    }
    s[n] = '\0';
    
    printf("%s\n", s);
    
    assert(s[0] != '0' && strlen(s) == n && (atoi(s) % t == 0)); // POST: number is n digits, > 0 and divisible by t
    return 0;
}